infra/modules/azure_function_app/locals.tf

```diff
@@ -31,7 +31,7 @@ locals {
   }
 
   application_insights = {
-    enable = var.application_insights_connection_string != null
+    enable = nonsensitive(var.application_insights_connection_string) != null
```

Comment:
isn't

```suggestion
    enable = nonsensitive(var.application_insights_connection_string != null)
```

don't we risk to expose the connection string otherwise?

Reply:
The resulting boolean value is calculated at runtime, then it will not be stored anywhere.

I haven't verified, but I believe the difference between mine and your code is that whether the `enable` value is stored in the state file or not

Reply:
I'm unsure about marking a sensitive value as 'nonsensitive'

Reply:
Yes still working